### PR TITLE
[Merged by Bors] - chore: restore `cc` tactic

### PR DESCRIPTION
**Pull request by Komyyy**
The `cc` tactic is a high-cost tactic, so I restored `cc` tactics only if a proof can be reduced significantly.
I made sure that `cc` tactic works on in the all changed proofs except when the other specification changes in Lean 4 changes a goal.

---

- [x] depends on: #13403

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by grunweg** [comment]
!bench

File: Mathlib/Tactic/CC/Datatypes.lean [hunk 1]
```diff
@@ -130,7 +130,7 @@ structure CCConfig where
   /-- If `true`, then use excluded middle -/
   em : Bool := true
   /-- If `true`, we treat values as atomic symbols -/
-  values : Bool := true
+  values : Bool := false
```
Comment (joneugster) [inline Mathlib/Tactic/CC/Datatypes.lean:133]: Can you explain here quickly why you decided to change this?

**Review by joneugster — APPROVED** [review]
Looks good to me.

File: Mathlib/Tactic/CC/Datatypes.lean (hunk 1 above)
Comment (Komyyy) [reply Mathlib/Tactic/CC/Datatypes.lean:133]: This is a change in the dependent PR; I merge to master now...

Comment (Komyyy) [reply Mathlib/Tactic/CC/Datatypes.lean]: @joneugster Merged!

**Review by Ruben-VandeVelde — APPROVED** [review]
This seems reasonable

maintainer merge

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**